web_timeline/static/src/js/timeline_renderer.js

```diff
@@ -308,41 +308,164 @@ odoo.define('web_timeline.TimelineRenderer', function (require) {
+                        content: gr_name[1],
+                    })
+                    self.$select_groups.append($("<option value='" + gr_name[0] + "' selected='selected'>" + gr_name[1] + "</option>"));
                 });
-                return self.on_data_loaded_2(nevents, group_bys, adjust_window);
+                self.$select_groups.multiselect('rebuild');
+                self.timeline.setGroups();
+                self.updateSelectedGroups();
+                return true;
+            });
+        },
+        arrayRemove: function (arr, id) {
+           return arr.filter( function (elmt) {
+               return elmt.id != id;
+           });
+        },
+        updateSelectedGroups: function() {
+            var self = this;
+            var selected = self.$select_groups.find("option:selected");
+            self.selected_groups = [];
+            _.each(selected, function (elmt) {
+                self.selected_groups.push({
+                    id: elmt.value,
+                    content: elmt.text,
+                })
+            });
+            self.timeline.setGroups(self.selected_groups);
+        },
+        createSelectGroups: function () {
+            var self = this;
+            var select_groups = $(core.qweb.render('TimelineSelectGroups', {'groups': self.groups}));
+            self.$el.find('.selected-groups').html(select_groups);
+            self.$select_groups = select_groups;
+            select_groups.multiselect({
+                buttonWidth: '350px',
+                maxHeight: 400,
+                enableFiltering: true,
+                enableClickableOptGroups: true,
+                includeSelectAllOption: true,
+                allSelectedText: _t('All groups selected'),
+                onChange: function(element, checked) {
+                    self.updateSelectedGroups();
+                },
+                onSelectAll: function() {
+                    self.updateSelectedGroups();
+                },
+                onDeselectAll: function() {
+                    self.updateSelectedGroups();
+                },
             });
         },
-
         /**
          * Set groups and events.
          *
          * @private
          */
-        on_data_loaded_2: function (events, group_bys, adjust_window) {
+        on_data_loaded_2: function (events, group_bys, x2x, adjust_window) {
             var self = this;
             var data = [];
             var groups = [];
             this.grouped_by = group_bys;
             _.each(events, function (event) {
                 if (event[self.date_start]) {
-                    data.push(self.event_data_transform(event));
+                    if (x2x) {
+                        _.each(event[group_bys], function (gr) {
+                            var x2x_object = jQuery.extend({}, event);
+                            x2x_object[group_bys] = [gr];
+                            x2x_object.id = event.id + "_" + gr;
```

Comment:
@Tardo Same comment as previously. The id I generate is alredy unique (due to the uniq relationship between record in the one2many and many2many relationship).
In addition to that it will be much more easier to debug when you know that your id is id + _ + group than something completely random

Reply:
Ok, can you export it to a function? More readable if you now that it's an unique id... how do you see it?

Reply:
For the moment there is no use of the "gr" in the id_gr. It is only for debugging purpose but I also think that it is the most precise id for this situation. this element represent the record with id = id in the group with id = gr.